library/core/src/ops/index.rs

```diff
@@ -163,6 +163,12 @@ see chapter in The Book <https://doc.rust-lang.org/book/ch08-02-strings.html#ind
 #[doc(alias = "[")]
 #[doc(alias = "]")]
 #[doc(alias = "[]")]
+#[doc(alias = "[ ]")]
+#[doc(alias = "[0]")]
+#[doc(alias = "[1]")]
+#[doc(alias = "[i]")]
+#[doc(alias = "[j]")]
+#[doc(alias = "[n]")]
```

Comment:
I am a little surprised to see `[` and `]` as aliases here, and the 'random' indices (0, 1) and characters here also seem a little bit going too far. In practice, if you know that what you want is indexing, you'll probably look for "index", right?

And if not, then I imagine it won't occur to most people to use the standard library docs with `[1]` or similar -- and I don't know that we should encourage that. If you don't really know what `[1]` does, this is probably not the ideal place to land -- something like https://doc.rust-lang.org/nightly/reference/expressions/array-expr.html might be preferable, or even https://doc.rust-lang.org/nightly/book/ch03-02-data-types.html#accessing-array-elements. In any case, it seems a rare case, and feels like the wrong use for aliases to me.

Reply:
I agree, these probably go to far. I was doing them because `array` had `[T; N]` and `[T;N]` synonyms with seem equally questionable IMO. Who says the _type_ is called `T` and the _number_ called `N`? So I’ve added `[T; 0]` and `[T; 1]` in an effort to get more realistic small numbers, and for indexing there wasn’t really a canonical variable name, so `i`, `j` and `n` seemed reasonable.

How much shall be removed then? Are `[T; N]` and `[T;N]` for `array`s really wanted? Should all the symbolic aliases be removed from `Index` and `IndexMut` be removed (and consequently also the `(` and `)` -based aliases for `FnOnce` that I’ve added)?

Reply:
OTOH, if we get a fuzzy search for aliases, terms like ` [` or `[;]` could find an alias `[T; N]` which would in and by itself then be a better preview of the syntax that this is actually about. One could throw in a `[a, b, c]` alias for array expressions, too. Similarly, indexing could be something like `a[i]` and function-call expressions could be `f(x)` or `f(x, y, z)`.